src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs

```diff
@@ -552,6 +558,67 @@ internal static void VerifyNoNullability(Symbol symbol)
             AssertEx.Equal(ImmutableArray<Symbol>.Empty, symbols);
         }
 
+        private static void VerifyTypes(CSharpCompilation compilation, SyntaxTree tree)
+        {
+            var root = tree.GetRoot();
+            var annotations = GetAnnotations(root);
+            if (annotations.IsEmpty)
+            {
+                return;
+            }
+            var model = compilation.GetSemanticModel(tree);
+            var expectedTypes = annotations.SelectAsArray(annotation => annotation.Item2);
+            var actualTypes = annotations.SelectAsArray(annotation => ((CSharpDataFlowAnalysis)model.AnalyzeDataFlow(annotation.Item1)).TypeAndNullability.GetDebuggerDisplay());
+            AssertEx.Equal(expectedTypes, actualTypes);
+        }
+
+        private static ImmutableArray<(ExpressionSyntax, string)> GetAnnotations(SyntaxNode root)
```

Comment:
Consider naming the tuple elements (rather than Item1/Item2).

Reply:
Done.